src/hooks/useAlert.js

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useRef, useState } from 'react';
+
+export const useAlert = (initialAlert = { type: '', message: '' }) => {
+  // const [listNotFound, setListNotFound] = useState('');
+  // const [error, setError] = useState('');
+  const [load, setLoad] = useState(initialAlert);
+
+  const [alertMessage, setAlertMessage] = useState(initialAlert);
+  const timeoutIdRef = useRef();
+
+  const setTimeoutAlert = () => {
+    timeoutIdRef.current = setTimeout(() => {
+      setAlertMessage(initialAlert);
+    }, 2000);
+  };
+
+  useEffect(() => {
+    return () => {
+      if (timeoutIdRef.current) clearTimeout(timeoutIdRef.current);
+    };
+  }, []);
+
+  return [alertMessage, setTimeoutAlert, setAlertMessage, load, setLoad];
```

Comment:
You may want to return these as an object rather than array.
```javascript
return { alertMessage, setTimeoutAlert, setAlertMessage, load, setLoad };
```
 That would make it easier to consume. Imagine that we want to use this hook in a component and only want to use `setLoad`. If it's an array, then we'd have to access the 5th element in the array somehow. If it was an object, then we could just do something like:

```javascript
const { setLoad } = useAlert();
```